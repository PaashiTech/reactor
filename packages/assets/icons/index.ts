import CheckGreen from "./svg/check-green.svg";
import Plus from "./svg/plus-icon.svg";
import Edit from "./svg/edit.svg";
import ChevronLeft from "./svg/chevron-left.svg";
import Warning from "./svg/warning.svg";
import ToastLogo from "./svg/toast-logo.svg";

export { CheckGreen, Plus, Edit, ChevronLeft, Warning, ToastLogo };
