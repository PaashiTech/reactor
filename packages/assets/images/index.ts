import IntroToAA from "./intro-to-aa.png";
import Carousal1 from "./carousal-1.png";
import Carousal2 from "./carousal-2.png";
import Carousal3 from "./carousal-3.png";
import Carousal4 from "./carousal-4.png";

export { IntroToAA, Carousal1, Carousal2, Carousal3, Carousal4 };
