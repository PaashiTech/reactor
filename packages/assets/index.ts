/**
 * The package that provides assets (icons, fonts, gifs etc.) to all the packages.
 *
 * @package assets
 */

// Fonts
export * from "./fonts";

// Icons
export * as unmzIcons from "./icons";
