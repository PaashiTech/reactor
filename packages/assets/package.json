{
  "name": "@unmaze/assets",
  "version": "0.0.1",
  "private": true,
  "main": "src/index.ts",
  "dependencies": {
    "@tamagui/font-inter": "1.89.3",
    "expo-font": "11.10.2"
  },
  "peerDependencies": {
    "@babel/core": "^7.23.9"
  }
}
