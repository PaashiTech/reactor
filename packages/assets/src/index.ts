/**
 * The package that provides assets (icons, fonts, gifs etc.) to all the packages.
 *
 * @package assets
 */

// Fonts
export * from "../fonts";

// Icons
export * from "../icons";

// Images
export * from "../images";
