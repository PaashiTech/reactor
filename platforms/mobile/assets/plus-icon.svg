<svg width="21" height="20" viewBox="0 0 21 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M4.66663 10H16.3333" stroke="#393939" stroke-width="1.67" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M10.5 4.16667V15.8333" stroke="#393939" stroke-width="1.67" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
