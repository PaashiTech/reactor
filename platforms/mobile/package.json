{
  "name": "@unmaze/mobile",
  "private": true,
  "types": "./src",
  "main": "src/index.tsx",
  "files": [
    "types",
    "dist"
  ],
  "sideEffects": [
    "*.css"
  ],
  "scripts": {
    "web": "yarn dev",
    "dev:vite": "NODE_ENV=development vite --host --clearScreen false",
    "dev:vite:extract": "EXTRACT=1 yarn dev",
    "dev": "yarn dev:webpack",
    "dev:webpack": "DISABLE_EXTRACTION=true NODE_ENV=development webpack serve",
    "dev:extract": "DISABLE_EXTRACTION=false NODE_ENV=development webpack serve",
    "prod": "NODE_ENV=production webpack --json=dist/compilation-stats.json",
    "upgrade:tamagui": "yarn up '*tamagui*'@latest '@tamagui/*'@latest"
  },
  "dependencies": {
    "@react-navigation/native": "^6.1.9",
    "@react-navigation/native-stack": "^6.9.17",
    "@unmaze/views": "workspace:*",
    "@unmaze/config": "workspace:*",
    "@unmaze/assets": "workspace:*",
    "react": "^18.2.0",
    "react-native": "0.73.3",
    "react-native-safe-area-context": "4.8.2",
    "react-native-screens": "3.29.0"
  },
  "devDependencies": {
    "babel-polyfill": "^6.26.0",
    "esbuild-loader": "^4.0.2",
    "esbuild-register": "^3.4.2",
    "react-refresh": "^0.14.0"
  }
}
