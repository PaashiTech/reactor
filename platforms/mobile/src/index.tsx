export { default as App } from "./App";

export * from "@unmaze/views";
// export * from "@unmaze/config";
