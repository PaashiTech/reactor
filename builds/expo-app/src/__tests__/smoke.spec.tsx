import React from "react";
import renderer from "react-test-renderer";
import { View, Text } from "react-native";
import {
  View as TamaguiView,
  Text as TamaguiText,
  TamaguiProvider,
  tamaguiConfig,
} from "@unmaze/mobile";
import { App } from "@unmaze/mobile";

import { NavigationContainer } from "@react-navigation/native";
import { useUnmzFontsExpo } from "@unmaze/assets";

const AppWithNavigator = () => {
  const [fontsLoaded] = useUnmzFontsExpo();

  if (!fontsLoaded) {
    return null;
  }

  return (
    <NavigationContainer>
      <App />
    </NavigationContainer>
  );
};

const SimpleView = () => {
  return (
    <View>
      <Text>RN Test</Text>
    </View>
  );
};

const SimpleTamaguiView = () => {
  return (
    <TamaguiProvider config={tamaguiConfig} defaultTheme="light">
      <TamaguiView>
        <TamaguiText>Tamagui Test</TamaguiText>
      </TamaguiView>
    </TamaguiProvider>
  );
};

describe("Rendering test suit", () => {
  test("renders simple RN views and texts correctly", () => {
    const tree = renderer.create(<SimpleView />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders simple Tamagui views and texts correctly", () => {
    const tree = renderer.create(<SimpleTamaguiView />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test.skip("App renders correctly", () => {
    const tree = renderer.create(<AppWithNavigator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
