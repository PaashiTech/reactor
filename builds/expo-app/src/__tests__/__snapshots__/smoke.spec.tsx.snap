// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test suit renders simple RN views and texts correctly 1`] = `
<View>
  <Text>
    RN Test
  </Text>
</View>
`;

exports[`Rendering test suit renders simple Tamagui views and texts correctly 1`] = `
<View
  style={{}}
>
  <Text
    style={
      {
        "color": "hsl(0, 0%, 9.0%)",
        "fontFamily": "Satoshi",
      }
    }
    suppressHighlighting={true}
  >
    Tamagui Test
  </Text>
</View>
`;
